include/iterlib/RandomIterator.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+// get random results from Iterator
+class RandomIterator : public WrappedIterator {
+public:
+  RandomIterator(Iterator* iter, int count)
+    : WrappedIterator(iter)
+    , count_(count)
```

Comment:
Can you add this in the comments : 
https://en.wikipedia.org/wiki/Reservoir_sampling
Also, mention what happens if the inner iterator has lesser elements than count?